include/pcp/kdtree/linked_kdtree.hpp

```diff
@@ -98,27 +126,72 @@ class basic_linked_kdtree_t
         }
     }
 
+    /**
+     * @brief Checks if kdtree is empty
+     * @return True if kdtree is empty
+     * 
+    */
     bool empty() const { return storage_.empty(); }
+
+    /**
+     * @brief Number of elements in the kdtree
+     * @return Number of elements in the kdtree 
+    */
     std::size_t size() const { return storage_.size(); }
+
+    /**
+     * @brief Removes all elements from the kdtree
+    */
     void clear()
     {
         root_.reset();
         storage_.clear();
     }
 
+    /**
+     * @brief Iterator to the first element of this kdtree
+     * @return Iterator to the first element of this kdtree 
+    */
     iterator begin() { return storage_.begin(); }
+
+    /**
+     * @brief End iterator to this kdtree's elements
+     * @return  End iterator to this kdtree's elements
+    */
     iterator end() { return storage_.end(); }
+
+    /**
+     * @brief Const iterator to the first element of this ktree
+     * @return Const iterator to the first element of this kdtree
+    */
     const_iterator cbegin() const { return storage_.cbegin(); }
     const_iterator cend() const { return storage_.cend(); }
 
+    /**
+     * @brief Root of the kdtree
+     * @return Root of the kdtree
+    */
     node_type_ptr const& root() const { return root_; }
+    /**
+     * @brief Bounding box of the kdtree
+     * @return Bounding box of the kdtree 
+    */
     aabb_type const& aabb() const { return aabb_; }
 
+    /**
+     * @brief Returns the k-nearest-neighbours in K dimensions Euclidean space 
```

Comment:
Maybe specify that we won't return points that are duplicated of the target point.